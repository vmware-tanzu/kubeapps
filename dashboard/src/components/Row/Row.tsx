// Copyright 2021-2023 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import { HTMLAttributes } from "react";

export interface IRowProps {
  isList?: boolean;
  children: React.ReactNode;
}

export function Row({ isList, children }: IRowProps) {
  const innerProps: HTMLAttributes<HTMLDivElement> = {
    className: "clr-row",
    role: isList ? "list" : "",
  };

  return <div {...innerProps}>{children}</div>;
}
