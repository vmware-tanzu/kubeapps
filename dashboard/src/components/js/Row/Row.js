// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import PropTypes from "prop-types";
import React from "react";

/**
 * https://v2.clarity.design/grid
 */
const Row = ({ list, children }) => {
  const innerProps = {
    className: "clr-row",
  };

  if (list === true) {
    innerProps["role"] = "list";
  }

  return <div {...innerProps}>{children}</div>;
};

Row.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.element,
    PropTypes.string,
  ]).isRequired,
  // For accessibility. A lot of times, rows include a list of elements
  list: PropTypes.bool,
};

Row.defaultProps = {
  list: false,
};

export default Row;
