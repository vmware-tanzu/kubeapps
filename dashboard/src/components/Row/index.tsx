// Copyright 2020-2023 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import { Row } from "./Row";

export default Row;
