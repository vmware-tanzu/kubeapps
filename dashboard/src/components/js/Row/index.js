// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import Row from "./Row";
export default Row;
