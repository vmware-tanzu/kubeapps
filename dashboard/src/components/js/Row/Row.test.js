// Copyright 2020-2022 the Kubeapps contributors.
// SPDX-License-Identifier: Apache-2.0

import { shallow } from "enzyme";
import React from "react";
import Row from ".";

describe(Row, () => {
  it("renders the content correctly", () => {
    const text = "My Text";
    const wrapper = shallow(<Row>{text}</Row>);

    expect(wrapper).toHaveText(text);
    expect(wrapper).toMatchSnapshot();
  });

  it("includes the expected CSS class", () => {
    const wrapper = shallow(<Row>Test</Row>);
    expect(wrapper).toHaveClassName("clr-row");
  });

  it("add the role when the row is a list", () => {
    const wrapper = shallow(<Row list>Test</Row>);
    expect(wrapper.prop("role")).toBe("list");
  });
});
