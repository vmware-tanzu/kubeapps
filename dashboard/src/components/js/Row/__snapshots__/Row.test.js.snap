// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders the content correctly 1`] = `
<div
  className="clr-row"
>
  My Text
</div>
`;
